chainerx_cc/chainerx/native/native_device/fill.cc

```diff
@@ -159,6 +159,28 @@ class NativeFillKernel : public FillKernel {
 
 CHAINERX_NATIVE_REGISTER_KERNEL(FillKernel, NativeFillKernel);
 
+class NativeTriKernel : public TriKernel {
+public:
+    void Call(int64_t k, const Array& out) override {
+        VisitDtype(out.dtype(), [k, &out](auto pt) {
+            using T = typename decltype(pt)::type;
+            struct Impl {
+                void operator()(int64_t i, T& out) {
+                    int64_t row = i % m;
+                    int64_t col = i / m;
+                    out = row <= col + k ? T{1} : T{0};
```

Comment:
A minor but I think the rows and cols are mixed up (the logic seem correct, but the names).

Reply:
Indeed they are mixed up. I took the kernel from [CuPy.](https://github.com/cupy/cupy/blob/v6.2.0/cupy/creation/matrix.py#L84)

Reply:
Indeed they are mixed up. I took the kernel from [CuPy.](https://github.com/cupy/cupy/blob/v6.2.0/cupy/creation/matrix.py#L84)

Reply:
Indeed they are mixed up. I took the kernel from [CuPy.](https://github.com/cupy/cupy/blob/v6.2.0/cupy/creation/matrix.py#L84)

Reply:
Thanks, I also noticed it was wrong in CuPy (and NumPy's docs) so I sent a fix.